### ISSUE TITLE
Add support for using system PATH installed dependencies

### ISSUE THREAD

**Issue by pzelnip** [chrissimpkins/Crunch#40]
Rather than compiling from source it'd be nice if Crunch could just make use of brew-installed versions of pngquant & zopflipng.

**Comment by chrissimpkins** [comment]
Thanks for the suggestion and PR Adam.  This was an intentional decision to fix the dependency versions rather than use package-managed versions of pngquant and zopflipng.  Default support for user-defined versions of the dependencies makes testing against the dependency releases across the range of Homebrew and Linux package managers more difficult, potentially eliminates the validity of the data that are displayed to users on the README page due to the non-deterministic file output at a given release of the Crunch applications, and means that when/if something breaks due to changes in the dependencies down the road (for instance a major version release in one of the two tools), the issue report winds up here.

I do intend to keep this project current with releases of both dependencies when new changes occur that will influence file sizes or address bugs (see https://github.com/chrissimpkins/Crunch/issues/34), and test against these changes before releasing the applications for use.    I just don't feel terribly compelled to support always current versions as defined by the user.  The executable binaries for both pngquant and zopflipng are both under 1MB in size so it is not a significant issue to have multiple installed copies on a machine in my opinion and the compiled files in this project are stored off of PATH so it doesn't restrict one's ability to install other, more current versions of the tools for other uses.  I guess the benefit here is that you avoid the 30-45 sec source compile process. 

Since the intent of your report is to eliminate source compiles for macOS users, what are your thoughts about falling back to the embedded Crunch macOS GUI versions of pngquant and zopflipng if they are available on the user's system?  Once the GUI tool is installed they are on the paths:

```
/Applications/Crunch.app/Contents/Resources/pngquant
/Applications/Crunch.app/Contents/Resources/zopflipng
```

The Crunch GUI can be installed through Homebrew cask so the install approach on macOS would then be:

```
$ homebrew cask install crunch
$ git clone https://github.com/chrissimpkins/Crunch.git
$ cd Crunch
$ make install-executable
```

The change c/w your suggested approach is that you swap the Homebrew cask install for the Homebrew pngquant / zopflipng installs.  All of this could be placed in an install script if that is helpful.

Thoughts?  Open to further discussion about this.  @perillamint also suggested this for Linux distros.

**Comment by pzelnip** [comment]
That's totally fair.

My intent with the issue & change was to make setup easier for users who already have brew installed the underlying tools.  I already had pngquant & zopflipng installed by homebrew, so I didn't want to install them again, but rather use the tools I already had installed (and want to continue to leave installed).  I also had no interest in a GUI component, as I only want to use this tool from the command-line.  It was less about avoiding compiling, and more about using what I already had.

Basically what happened was I found this project, and modified a local copy of crunch.py to do this to make it work with what I already had setup, and then thought it might be useful for others.  Totally cool if you want to close this.

**Comment by chrissimpkins** [comment]
How about we add the paths as defined currently as constants at the top of the module so that anyone who wants to modify it can easily access those and redefine them to system installed versions without digging into the functions in the module.  We can return the constants from the get_* functions when the script is being used from the command line so whatever is defined there are the versions that are used.  How does that sound?   Can update docs with this info for those who want to tinker with the source.

**Comment by chrissimpkins** [comment]
Changes suggested in https://github.com/chrissimpkins/Crunch/issues/40#issuecomment-390066703 were added and pushed to pzelnip-homebrew branch.

**Comment by chrissimpkins** [comment]
PR #41 remains open.  Adam, feel free to continue to modify there if you have additional thoughts and I will rebase your pull request on the `pzelnip-homebrew` branch.

**Comment by chrissimpkins** [comment]
These changes will be part of the next release

**Comment by chrissimpkins** [comment]
released in v2.1.0
